instaloader/nodeiterator.py

```diff
@@ -318,12 +319,14 @@ def resumable_iteration(context: InstaloaderContext,
             context.error("Warning: Not resuming from {}: {}".format(resume_file_path, exc))
     try:
         yield is_resuming, start_index
-    except (KeyboardInterrupt, AbortDownloadException):
+    except (KeyboardInterrupt, AbortDownloadException, QueryReturnedBadRequestException):
```

Comment:
QueryReturnedBadRequestException is too specific, I think it would be beneficial to save the resume information for other errors as well.

I suggest to instead catch InstaloaderException here, to save the resume information always when the loop body is aborted due to an error.